File: src/main/java/net/dontdrinkandroot/example/angularrestspringsecurity/config/RestConfiguration.java
package net.dontdrinkandroot.example.angularrestspringsecurity.config;

import net.dontdrinkandroot.example.angularrestspringsecurity.entity.BlogPost;
import net.dontdrinkandroot.example.angularrestspringsecurity.entity.User;
import net.dontdrinkandroot.example.angularrestspringsecurity.repository.BlogPostRepository;
import net.dontdrinkandroot.example.angularrestspringsecurity.repository.UserRepository;
import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurerAdapter;
import org.springframework.stereotype.Component;

@Component
public class RestConfiguration extends RepositoryRestConfigurerAdapter
{
    @Override
    public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config)
    {
        config.withEntityLookup()
                .forRepository(UserRepository.class, User::getUsername, UserRepository::findByUsername)
                .forRepository(BlogPostRepository.class, BlogPost::getSlug, BlogPostRepository::findBySlug);
    }
}

File: src/main/kotlin/net/dontdrinkandroot/example/angularrestspringsecurity/config/RestConfiguration.kt
package net.dontdrinkandroot.example.angularrestspringsecurity.config

import net.dontdrinkandroot.example.angularrestspringsecurity.entity.BlogPost
import net.dontdrinkandroot.example.angularrestspringsecurity.entity.User
import net.dontdrinkandroot.example.angularrestspringsecurity.repository.BlogPostRepository
import net.dontdrinkandroot.example.angularrestspringsecurity.repository.UserRepository
import org.springframework.context.annotation.Configuration
import org.springframework.data.rest.core.config.RepositoryRestConfiguration
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Configuration
class RestConfiguration : RepositoryRestConfigurer {

    override fun configureRepositoryRestConfiguration(config: RepositoryRestConfiguration) {
        config.withEntityLookup()
                .forRepository(UserRepository::class.java, User::username, UserRepository::findByUsername)
                .forRepository(BlogPostRepository::class.java, BlogPost::slug, BlogPostRepository::findBySlug)
    }
}
